packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -702,4 +703,38 @@ describeSpec('Listens:', [], () => {
         .enableNetwork();
     }
   );
+
+  specTest(
+    'Query is re-listened to after primary tab failover',
+    ['multi-client'],
+    () => {
+      const query = Query.atPath(path('collection'));
+      const docA = doc('collection/a', 1000, { key: 'a' });
+      const docB = doc('collection/b', 2000, { key: 'a' });
+
+      return client(0, false)
+        .expectPrimaryState(true)
+        .client(1)
+        .userListens(query)
+        .expectEvents(query, { fromCache: true })
+        .client(0)
+        .expectListen(query)
+        .watchAcksFull(query, 1000, docA)
+        .client(1)
+        .expectEvents(query, { added: [docA] })
+        .client(2)
+        .userListens(query)
+        .expectEvents(query, { added: [docA] })
+        .client(0)
+        .shutdown()
+        .client(1)
+        .runTimer(TimerId.ClientMetadataRefresh)
```

Comment:
Per my other comments, I think you need a test where e.g. client(2) has a listen that client(1) does not...

Reply:
Agreed! Added a test for this below (`Query is established in new primary tab`).